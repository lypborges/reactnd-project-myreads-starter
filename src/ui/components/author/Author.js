import React from "react";
import PropTypes from "prop-types";

const Author = props => {
  let element = <span />;
  const authors = props.authors;
  if (authors && authors.length > 0) {
    element = props.authors.map(author => {
      return <span key={author}>{author}</span>;
    });
  }
  return element;
};

Author.propTypes = {
  authors: PropTypes.array
};

export default Author;
