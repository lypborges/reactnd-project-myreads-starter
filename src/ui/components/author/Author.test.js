import React from "react";
import renderer from "react-test-renderer";
import Author from "./Author";

describe("Author layout", () => {
  test("render with authors", () => {
    const authors = ["William E. Shotts, Jr.", "Felipe Borges Ferreira"];
    const tree = renderer.create(<Author authors={authors} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("render without authors", () => {
    const tree = renderer.create(<Author />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
