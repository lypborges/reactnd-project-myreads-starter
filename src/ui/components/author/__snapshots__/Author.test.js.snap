// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author layout render with authors 1`] = `
Array [
  <span>
    William E. Shotts, Jr.
</span>,
  <span>
    Felipe Borges Ferreira
</span>,
]
`;

exports[`Author layout render without authors 1`] = `<span />`;
